aca_lts_eval.py

```diff
@@ -176,10 +177,45 @@ def select_ri_guide_stars(ra, dec, cone_stars):
     return G_RI_CAT_CACHE[id_key]
 
 
-def t_lose_star(mag, ref_mag=10.3, ref_tccd=-11.5, scale_4c=None):
+def guide_count(mags, tccd=-11.5):
+    """
+    Given mags from guide stars and a temperature, calculate a guide star
+    count using signal-to-noise scaled mag thresholds.
+    """
+    thresh1 = snr_mag_for_tccd(tccd, ref_mag=10.0)
+    thresh2 = snr_mag_for_tccd(tccd, ref_mag=10.2)
+    thresh3 = snr_mag_for_tccd(tccd, ref_mag=10.3)
+    counts = np.zeros_like(mags)
+    counts[mags <= thresh1] = 1.0
+    counts[(mags <= thresh2) & (mags > thresh1)] = 0.75
+    counts[(mags <= thresh3) & (mags > thresh2)] = 0.5
+    return np.sum(counts)
+
+
+def snr_mag_for_tccd(tccd, ref_mag=10.3, ref_tccd=-11.5, scale_4c=None):
+    """
+    Given a tccd, solve for the magnitude that has the same expected signal
+    to noise as ref_mag / ref_tccd.
+    """
     if scale_4c is None:
         scale_4c = dark_model.DARK_SCALE_4C
-    return (1.6 * (ref_mag - mag)) / np.log10(scale_4c) + ref_tccd
+    return ref_mag - (tccd - ref_tccd) * np.log10(scale_4c) / 1.6
+
+
+def t_ccd_for_guide(mags, min_guide_count=4, warm_t_ccd=-5, cold_t_ccd=-16):
+    def n_gui_above_min(t_ccd):
+        count = guide_count(mags, t_ccd)
+        return count - min_guide_count
+    # In the style of chandra_aca.star_probs.t_ccd_warm_limit, use an optimization
+    # strategy to solve for the warmest temperature that still gets the min_guide_count
+    merit_func = n_gui_above_min
+    if merit_func(warm_t_ccd) >= 0:
+        t_ccd = warm_t_ccd
+    elif merit_func(cold_t_ccd) <= 0:
+        t_ccd = cold_t_ccd
+    else:
+        t_ccd = bisect(merit_func, cold_t_ccd, warm_t_ccd, xtol=1e-4, rtol=1e-4)
```

Comment:
The step-wise nature of the guide count didn't seem to behave well with the brentq optimizer used in t_ccd_warm_limit.